File: go/design-patterns/bridge/concrete-devices/concrete_devices.go
type Tv struct {
    on     bool
    volume int
}
func (t *Tv) IsEnabled() bool { return t.on }
func (t *Tv) Enable()         { t.on = true }
func (t *Tv) Disable()        { t.on = false }
func (t *Tv) GetVolume() int  { return t.volume }
func (t *Tv) SetVolume(p int) { t.volume = p }

File: typescript/design-patterns/bridge/concrete-devices/ConcreteDevices.ts
class Tv implements Device {
  private on = false;
  private volume = 30;

  isEnabled() { return this.on; }
  enable() { this.on = true; }
  disable() { this.on = false; }
  getVolume() { return this.volume; }
  setVolume(p: number) { this.volume = Math.max(0, Math.min(100, p)); }
}

class Radio implements Device {
  private on = false;
  private volume = 15;

  isEnabled() { return this.on; }
  enable() { this.on = true; }
  disable() { this.on = false; }
  getVolume() { return this.volume; }
  setVolume(p: number) { this.volume = Math.max(0, Math.min(100, p)); }
}